tests/unit-core/test-arraybuffer.c

```diff
@@ -153,15 +153,37 @@ static void test_write_with_offset (uint8_t offset) /**< offset for buffer write
 
 static bool callback_called = false;
 static bool detach_free_callback_called = false;
+static bool user_callback_called = false;
```

Comment:
the type of this should be change to `int`, and the appropriate number of calls should be checked later. Same for `callback_called `